pkg/controller/controller.go

```diff
@@ -289,7 +339,21 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 		return nil, fmt.Errorf("claim Selector is not supported")
 	}
 
-	driverName, err := checkDriverState(p.grpcClient, p.timeout)
+	var checkSnapshotSupport bool
+	if options.PVC.Spec.DataSource != nil {
+		// PVC.Spec.DataSource.Name is the name of the VolumeSnapshot API object
+		if options.PVC.Spec.DataSource.Name == "" {
```

Comment:
Since validation code already check this, do we need these checks here?

Reply:
Actually these can be removed now.  I'll remove them.

Reply:
Maybe we should keep these checks here?  When we add support for more Data Source types, the validation check will be changed to allow other Data Sources.